### PR TITLE
Finalize 0.10.4

### LINKED ISSUE #1992: v0.10.4 release planning
Release v0.10.4 is scheduled for this Friday.  Here is a proposed release plan for discussion:

Include in v0.10.4:
- #1944 
- #1985
- #1995 

Defer to the future:
- #1689, #1690, #1962
- #1708
- #1763 
- #1781 
- #1893 
- #1979, pending agreement that the method is sufficiently well-defined in the reference

### PR DESCRIPTION
**Pull request by kandersolar**
- [x] Closes #1992
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - ~[ ] Tests added~
 - ~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~[ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Leaving as draft while reviewing the various CI jobs

### CODE REVIEW

**Comment by kandersolar** [comment]
Inspecting the test output, I see a few warnings like this:

```
  /Users/runner/work/pvlib-python/pvlib-python/pvlib/inverter.py:535: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
  You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
  A typical example is when you are setting values in a column of a DataFrame, like:
  
  df["col"][row_indexer] = value
  
  Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.
  
  See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
    coeffs['a'][d] = a
```

and many warnings like this:

```
  /Users/runner/work/pvlib-python/pvlib-python/pvlib/tests/test_temperature.py:257: FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead.
```

The former is coming from `inverter.py`, so inside pvlib itself.  I'm not immediately seeing a hard release date for pandas 3.0, but it might be as early as April 1 (https://github.com/pandas-dev/pandas/milestone/102), so before our next release.  I will submit a PR to fix those in this release.  

The latter is from the tests, so lesser priority to fix, but easy enough so might as well.

File: docs/sphinx/source/whatsnew/v0.10.4.rst [hunk 1]
```diff
@@ -1,14 +1,14 @@
 .. _whatsnew_01040:
 
 
-v0.10.4 (Anticipated March, 2024)
----------------------------------
+v0.10.4 (March 18, 2024)
```
Comment (AdamRJensen) [inline docs/sphinx/source/whatsnew/v0.10.4.rst:4]:
```suggestion
v0.10.4 (March 19, 2024)
```

**Review by AdamRJensen — COMMENTED** [review]
Remember to add ``docs/sphinx/source/whatsnew/v0.10.5.rst``

**Review by AdamRJensen — APPROVED** [review]

**Review by cwhanse — APPROVED** [review]

**Comment by kandersolar** [comment]
I'll proceed with making the release.  Many thanks as always to all contributors!

**Pull request merged**